go/vt/vtgate/planbuilder/operator_transformers.go

```diff
@@ -605,9 +605,7 @@ func generateInsertShardedQuery(ins *sqlparser.Insert) (prefix string, mids sqlp
 		ins.Table, ins.Columns)
 	prefix = prefixBuf.String()
 
-	suffixBuf := sqlparser.NewTrackedBuffer(dmlFormatter)
-	suffixBuf.Myprintf("%v", ins.OnDup)
-	suffix = suffixBuf.String()
+	suffix = ins.OnDup
```

Comment:
we need to walk the `onDup` and remove the database qualifier from it

Reply:
I use git cherry-pick 02bbbdd and it prompts me "bad revision '02bbbdd'". Where is the commit in the repository?😂

Reply:
I submitted a new version and made a judgment when removing database qualifier. Using your test case